### PR TITLE
Use NumPy style docstrings.

### LINKED ISSUE #278: Convert all docstrings to numpy style
As part of our overall docs overhaul (see glotzerlab/signac-docs#64), we want to convert our docstrings to numpy style (as decided in glotzerlab/signac-docs#74). The best automated tool I'm familiar with for this task is [pyment](https://github.com/dadadel/pyment). In addition to converting docstrings, it will also generate docstrings for functions, classes, etc that are missing docstrings entirely. However, the conversion will require significant manual review to ensure that all docstrings are converted correctly.

### PR DESCRIPTION
**Pull request by bdice**
## Description
Applied NumPy docstring format to the package.

## Motivation and Context
Resolves #278. Supersedes #316.

Depends on #389, which depends on #387.

## Types of Changes
- [x] Documentation update

## Checklist:
- [x] I am familiar with the [**Contributing Guidelines**](https://github.com/glotzerlab/signac-flow/blob/master/CONTRIBUTING.md).
- [x] I agree with the terms of the [**Contributor Agreement**](https://github.com/glotzerlab/signac-flow/blob/master/ContributorAgreement.md).
- [x] My name is on the [list of contributors](https://github.com/glotzerlab/signac-flow/blob/master/contributors.yaml).
- [x] My code follows the [code style guideline](https://github.com/glotzerlab/signac-flow/blob/master/CONTRIBUTING.md#code-style) of this project.
- [x] The changes introduced by this pull request are covered by existing or newly introduced tests.

If necessary:
- [x] I have updated the API documentation as part of the package doc-strings.
- [ ] I have created a separate pull request to update the [framework documentation](https://docs.signac.io/) on [signac-docs](https://github.com/glotzerlab/signac-docs) and linked it here.
- [x] I have updated the [changelog](https://github.com/glotzerlab/signac-flow/blob/master/changelog.txt).

### CODE REVIEW

**Comment by kidrahahjo** [comment]
Assuming this is going to merge into `next`, I'm going to edit the base branch from `master` to `next`

**Comment by bdice** [comment]
@glotzerlab/signac-committers I'd like to get this reviewed/merged before the weekend if possible. Then I'll work on benchmarking, profiling, and improving performance.

File: flow/directives.py [hunk 1]
```diff
@@ -196,7 +196,7 @@ def update(self, other, aggregate=False, jobs=None, parallel=False):
             The other set of directives.
         aggregate : bool
             Whether to combine directives according to serial/parallel rules.
-        jobs :
+        jobs
```
Comment (b-butler) [inline flow/directives.py:199]:
```suggestion
        jobs : :class:`signac.contrib.job.Job` or tuple of :class:`signac.contrib.job.Job`
```

File: flow/directives.py [hunk 2]
```diff
@@ -215,7 +215,7 @@ def evaluate(self, jobs):
 
         Parameters
         ----------
-        jobs :
+        jobs
```
Comment (b-butler) [inline flow/directives.py:218]:
```suggestion
        jobs : :class:`signac.contrib.job.Job` or tuple of :class:`signac.contrib.job.Job`
```

File: flow/project.py [hunk 3]
```diff
@@ -3566,43 +3607,38 @@ def _submit_operations(
...
-        :param \*\*kwargs:
-            Additional keyword arguments to be forwarded to the scheduler.
-        :return:
+            for testing the submission workflow. (Default value = False)
+        show_template_help : bool
+            Show information about available template variables and filters and
+            exit. (Default value = False)
+        env : :class:`~.ComputeEnvironment`
+            The environment to use for submission. Uses the environment defined
+            by the :class:`~.FlowProject` if None (Default value = None).
```
Comment (b-butler) [inline flow/project.py:3633]: This should be after `_id`.

File: flow/project.py [hunk 4]
```diff
@@ -3683,43 +3719,38 @@ def submit_operations(
...
```
Comment (b-butler) [inline flow/project.py:3745]: This should be after `_id`.

**Review by b-butler — COMMENTED** [review]
Looks good, the PR is so large I could have easily missed something. I left a few comments that can be addressed.

**Comment by bdice** [comment]
@b-butler Thanks for the review! I fixed the things you mentioned. I'm planning to merge this in about 24 hours.

File: flow/project.py [hunk 5]
```diff
@@ -1007,75 +1012,95 @@ def eligible(self, job, ignore_conditions=IgnoreConditions.NONE):
         A FlowGroup is eligible for execution if at least one of its
```
Comment (kidrahahjo) [inline flow/project.py:1012]:
```suggestion
        A :class:`~.FlowGroup` is eligible for execution if at least one of its
```
I saw this change in line 971 of the diff (`project.py`). I think this should be reflected everywhere else?

File: flow/project.py [hunk 6]
```diff
@@ -1007,75 +1012,95 @@ def eligible(self, job, ignore_conditions=IgnoreConditions.NONE):
...
-        :rtype:
-            bool
+
         """
         return self._eligible((job,), ignore_conditions)
 
     @deprecated(deprecated_in="0.11", removed_in="0.13", current_version=__version__)
     def complete(self, job):
-        """Check if all BaseFlowOperation post-conditions are met.
+        """Check if all BaseFlowOperation postconditions are met.
```
Comment (kidrahahjo) [inline flow/project.py:1034]:
```suggestion
        """Check if all :class:`~.BaseFlowOperation` postconditions are met.
```

File: flow/project.py [hunk 7]
```diff
@@ -1007,75 +1012,95 @@ def eligible(self, job, ignore_conditions=IgnoreConditions.NONE):
...
     def isdisjoint(self, group):
-        """Return whether two groups are disjoint (do not share any common operations).
+        """Return whether two groups are disjoint.
+
+        Groups are disjoint if they do not share any common operations.
 
-        :param group:
+        Parameters
+        ----------
+        group : :class:`flow.project.FlowGroup`
```
Comment (kidrahahjo) [inline flow/project.py:1074]: There is a little inconsistency with respect to the reference 
For example, :class:`~.FlowGroup` v/s :class:`~flow.project.FlowGroup`
Do we need to fix it?

**Review by kidrahahjo — COMMENTED** [review]
Great work @bdice :heart: :+1: 
Here are some very minor comments.

**Comment by bdice** [comment]
@kidrahahjo Thanks for the feedback, I fixed the issues you noted.

**Pull request merged**